static_analyzer/clustering/grouping.py

```
@@ -0,0 +1,576 @@
+    """Directories of the files a leaf cluster touches (its 'package')."""
+    return {os.path.dirname(path) for path in cluster_result.cluster_to_files.get(cid, set())}
+
+
+def _package_affinity(package: str, candidates: set[str]) -> int:
+    """Leading path segments *package* shares with its closest match in *candidates*."""
+    own = package.split(os.sep)
+    best = 0
+    for candidate in candidates:
+        other = candidate.split(os.sep)
+        shared = 0
+        for a, b in zip(own, other):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)